tests/clients/test_cases.py

```diff
@@ -0,0 +1,64 @@
+import pytest
+
+from py42.clients.cases import CasesClient
+from py42.services.cases import CasesService
+from py42.services.casesfileevents import CasesFileEventsService
+
+_TEST_CASE_NUMBER = 123456
+
+
+@pytest.fixture
+def mock_cases_service(mocker):
+    return mocker.MagicMock(spec=CasesService)
+
+
+@pytest.fixture
+def mock_cases_file_event_service(mocker):
+    return mocker.MagicMock(spec=CasesFileEventsService)
+
+
+class TestCasesClient:
+    def test_file_events_returns_cases_file_events_service(
+        self, mock_cases_service, mock_cases_file_event_service
+    ):
+        cases_client = CasesClient(mock_cases_service, mock_cases_file_event_service)
+        assert cases_client.file_events is mock_cases_file_event_service
+
+    def test_create_calls_cases_service_with_expected_params(
+        self, mock_cases_service, mock_cases_file_event_service
+    ):
+        cases_client = CasesClient(mock_cases_service, mock_cases_file_event_service)
+        cases_client.create("name", "subject", "assignee", "description", "observation")
```

Comment:
A helpful thing to do in python (that I have recently adopted) is to always use key-value arg specification (kwargs) when they are optional args. This tells the code-reader that they are optional.

Example:

```
cases_client.create("name", subject="subject", assignee="assignee", description="description", "observation")

mock_cases_service.create.assert_called_once_with(
            "name", subject="subject", assignee="assignee", description="description", observation="observation"
        )
```

Reply:
Changed.